### ISSUE TITLE
CrossEntropy size in chatbot does NOT align with the mask

### ISSUE THREAD

**Issue by aobo-y** [pytorch/tutorials#383]
PR https://github.com/pytorch/tutorials/pull/382

I feel how the following filter loss through mask is queationable
https://github.com/pytorch/tutorials/blob/master/beginner_source/chatbot_tutorial.py#L893-L894

assume `batch_size` is `64`, then
`mask` size is `(64)`
`crossEntropy` size is `(64, 1)`

I know you can take use of **[Broadcasting](https://pytorch.org/docs/stable/notes/broadcasting.html#broadcasting-semantics)**. That's why it can run without exceptions.
However, i guess Broadcasting changes `mask` to `(1, 64)` to work.

Below is sample test how it's wrong
<img width="425" alt="screenshot 2018-12-13 at 12 38 31 am" src="https://user-images.githubusercontent.com/5113450/49917996-73a7c580-fe6f-11e8-85f2-bd84397c63b4.png">

**Comment by soumith** [comment]
fixed via https://github.com/pytorch/tutorials/pull/382
